### ISSUE TITLE
Physics: Microstrip and stripline impedance calculator

### ISSUE THREAD

**Issue by rjwalters** [rjwalters/kicad-tools#350]
## Overview

Implement analytical transmission line calculations for microstrip and stripline geometries using the Hammerstad-Jensen equations.

## Parent Issue

Part of #348 Physics Module: Transmission Line Calculator

### Core Calculator

```python
# src/kicad_tools/physics/transmission_line.py

from dataclasses import dataclass
import math

@dataclass
class ImpedanceResult:
    """Result of impedance calculation."""
    z0: float              # Characteristic impedance (Ω)
    epsilon_eff: float     # Effective dielectric constant
    loss_db_per_m: float   # Total loss at reference frequency
    phase_velocity: float  # Signal velocity (m/s)
    
    @property
    def propagation_delay_ps_per_mm(self) -> float:
        """Propagation delay in ps/mm."""
        return 1e9 / self.phase_velocity  # Convert m/s to ps/mm

class TransmissionLine:
    """Analytical transmission line impedance calculator."""
    
    def __init__(self, stackup: Stackup):
        self.stackup = stackup
    
    def microstrip(
        self, 
        width_mm: float, 
        layer: str,
        frequency_ghz: float = 1.0,
    ) -> ImpedanceResult:
        """
        Calculate microstrip impedance.
        
        Uses Hammerstad-Jensen equations for Z0 and εeff.
        
        Args:
            width_mm: Trace width in mm
            layer: Layer name (e.g., "F.Cu")
            frequency_ghz: Frequency for loss calculation
            
        Returns:
            ImpedanceResult with Z0, εeff, loss, velocity
        """
        h = self.stackup.get_reference_plane_distance(layer)
        er = self.stackup.get_dielectric_above(layer).dielectric_constant
        t = self.stackup.get_copper_thickness(layer)
        
        return self._microstrip_calc(width_mm, h, er, t, frequency_ghz)
    
    def stripline(
        self,
        width_mm: float,
        layer: str,
        frequency_ghz: float = 1.0,
    ) -> ImpedanceResult:
        """
        Calculate stripline impedance (symmetric or asymmetric).
        
        Args:
            width_mm: Trace width in mm
            layer: Inner layer name (e.g., "In1.Cu")
            frequency_ghz: Frequency for loss calculation
        """
        # Get distances to both reference planes
        h1, h2 = self.stackup.get_stripline_geometry(layer)
        er = self.stackup.get_dielectric_constant(layer)
        t = self.stackup.get_copper_thickness(layer)
        
        return self._stripline_calc(width_mm, h1, h2, er, t, frequency_ghz)
    
    def width_for_impedance(
        self,
        z0_target: float,
        layer: str,
        mode: str = "auto",
        tolerance: float = 0.01,
    ) -> float:
        """
        Calculate trace width for target impedance.
        
        Args:
            z0_target: Target impedance in ohms
            layer: Layer name
            mode: "microstrip", "stripline", or "auto" (detect from layer position)
            tolerance: Convergence tolerance (default 1%)
            
        Returns:
            Trace width in mm
        """
        # Use Newton-Raphson or bisection to find width
```

### Hammerstad-Jensen Equations

```python
def _microstrip_calc(
    self, 
    w: float, 
    h: float, 
    er: float, 
    t: float,
    freq: float,
) -> ImpedanceResult:
    """Hammerstad-Jensen microstrip equations."""
    
    # Effective width (accounts for copper thickness)
    w_eff = w + (t / math.pi) * math.log(
        4 * math.e / math.sqrt((t / h) ** 2 + (t / (w * math.pi + 1.1 * t * math.pi)) ** 2)
    )
    
    # Effective dielectric constant
    u = w_eff / h
    a = 1 + (1/49) * math.log((u**4 + (u/52)**2) / (u**4 + 0.432)) + (1/18.7) * math.log(1 + (u/18.1)**3)
    b = 0.564 * ((er - 0.9) / (er + 3)) ** 0.053
    
    eps_eff = (er + 1) / 2 + ((er - 1) / 2) * (1 + 10 / u) ** (-a * b)
    
    # Characteristic impedance
    f = 6 + (2 * math.pi - 6) * math.exp(-(30.666 / u) ** 0.7528)
    z0 = (60 / math.sqrt(eps_eff)) * math.log(f / u + math.sqrt(1 + (2 / u) ** 2))
    
    # Phase velocity
    c = 299792458  # m/s
    v_p = c / math.sqrt(eps_eff)
    
    # Loss estimation (simplified)
    loss = self._estimate_loss(w, h, er, t, freq)
    
    return ImpedanceResult(
        z0=z0,
        epsilon_eff=eps_eff,
        loss_db_per_m=loss,
        phase_velocity=v_p,
    )
```

## Files to Create

- `src/kicad_tools/physics/transmission_line.py`

## Acceptance Criteria

- [ ] Microstrip Z0 calculation ±2% vs reference (Saturn PCB, TXLINE)
- [ ] Stripline Z0 calculation ±2% vs reference
- [ ] Effective dielectric constant calculation
- [ ] Conductor thickness correction
- [ ] Inverse calculation (Z0 → width) with <1% error
- [ ] Loss estimation (conductor + dielectric)
- [ ] Phase velocity / propagation delay
- [ ] Unit tests against known values

## Test Cases

```python
def test_microstrip_50ohm():
    """Standard 50Ω microstrip on FR4."""
    stackup = Stackup.jlcpcb_4layer()
    tl = TransmissionLine(stackup)
    
    # Known: ~8mil (0.2mm) width gives ~50Ω on 4.6mil prepreg
    result = tl.microstrip(width_mm=0.2, layer="F.Cu")
    assert result.z0 == pytest.approx(50, rel=0.05)

def test_width_for_50ohm():
    """Inverse calculation for 50Ω."""
    stackup = Stackup.jlcpcb_4layer()
    tl = TransmissionLine(stackup)
    
    width = tl.width_for_impedance(z0_target=50, layer="F.Cu")
    
    # Verify by forward calculation
    result = tl.microstrip(width_mm=width, layer="F.Cu")
    assert result.z0 == pytest.approx(50, rel=0.01)
```

## References

- Hammerstad & Jensen, "Accurate Models for Microstrip Computer-Aided Design", 1980
- IPC-2141A Section 4.2

## Related

- #348 Physics Module (parent)
- #349 Stackup parsing (dependency)
